c7n/resources/s3.py

```diff
@@ -1191,10 +1191,15 @@ def process_key(self, s3, key, bucket_name, info=None):
         if info is None:
             info = s3.head_object(Bucket=bucket_name, Key=k)
 
-        if 'ServerSideEncryption' in info:
-            if self.kms_id and info.get('SSEKMSKeyId', '') == self.kms_id:
-                return False
-            else:
+        # If the data is already encrypted with AES256 and this request is also
+        # for AES256 then we don't need to do anything
+        if info.get('ServerSideEncryption') == 'AES256' and not self.kms_id:
+            return False
```

Comment:
@kapilt - I did a little reading and I think this is how this should work - just wanted to confirm with you.  (I think it worked by accident in the previous code path, and now I am adding it explicitly)